tsdb/head.go

```diff
@@ -904,6 +909,52 @@ func (h *Head) Init(minValidTime int64) error {
 	return nil
 }
 
+// CleanRestart cleans up the WAL and Head chunks dir and restarts the Head.
+// It must not be called after calling head.Init()
+func (h *Head) CleanRestart() error {
```

Comment:
Please let me know if we need tests for this func.

Reply:
We will need a test for opening DB on checkpoint corruption. We don't need a test for `CleanRestart` specifically since it will get tested with corrupt checkpoint test.

Reply:
Note that CleanRestart should also clear what was replayed till now. I am adding a `resetInMemoryState` in https://github.com/prometheus/prometheus/pull/9185/ which you could re-use when it's merged.